swupd/packs.go

```diff
@@ -60,15 +60,21 @@ type PackInfo struct {
 	Warnings []string
 }
 
-// WritePack writes the pack of a given Manifest from a version to the version of the Manifest. The
-// outputDir is used to pick deltas and fullfiles. If not empty, chrootDir is tried first as a fast
-// alternative to decompressing the fullfiles.
-func WritePack(w io.Writer, m *Manifest, fromVersion uint32, outputDir, chrootDir string) (info *PackInfo, err error) {
-	if m.Name == "" {
+// WritePack writes the pack between two Manifests. The outputDir is used to pick deltas
+// and fullfiles. If not empty, chrootDir is tried first as a fast alternative to
+// decompressing the fullfiles.
+func WritePack(w io.Writer, fromManifest, toManifest *Manifest, outputDir, chrootDir string) (info *PackInfo, err error) {
+	if toManifest.Name == "" {
 		return nil, fmt.Errorf("manifest has no name")
 	}
-	if fromVersion >= m.Header.Version {
-		return nil, fmt.Errorf("fromVersion (%d) smaller than toVersion (%d)", fromVersion, m.Header.Version)
+	toVersion := toManifest.Header.Version
+
+	var fromVersion uint32
+	if fromManifest != nil {
+		fromVersion = fromManifest.Header.Version
+		if fromVersion >= toVersion {
+			return nil, fmt.Errorf("fromVersion (%d) smaller than toVersion (%d)", fromVersion, toVersion)
+		}
```

Comment:
You should also return if `fromManifest == nil`, that seems like an error case to me.

Reply:
"fromManifest == nil" implies we are creating a zeropack, so it is a valid usage. Added a sentence in the comments.

Also added a check for toManifest not being nil early on.